lib/linter.js

```diff
@@ -108,6 +109,14 @@ class Linter {
     return rule;
   }
 
+  findRulesForPath(overrides, filePath) {
```

Comment:
* please underscore (don't want to accidentally infer public api or anything)
* the name seems slightly off, we aren't finding the rules we are finding the specific configuration for this filePath
* Can you make a utility function (akin to `get-config`) that takes the current config and a filePath, and returns the "final config" for that file? That would replace this helper method, and much of the changes down in `buildRules` with something like `let configuredRules = getConfigForFile(this.config, options.filePath);`,  and we'd use `configuredRules` throughout (instead of `this.config`).